coralproject/talk
# Support response errors in the graphql framework

## What does this PR do?
- Add support for response errors to the graphql framework
  - Prevent execution of query reducers, and mutation updaters when there are response errors.
  - Fail mutation promises with a custom error `ResponseErrors` that returns an error message and an array of `ResponseError` at `ResponseErrors.errors`.
  - `ResponseErrors` and `ResponseError` implementation:
```js
class ResponseErrors extends Error {
  constructor(errors) {
    super(`Response Errors ${JSON.stringify(errors)}`);
    this.errors = errors.map((e) => new ResponseError(e));
  }
}

class ResponseError {
  constructor(error) {
    Object.assign(this, error);
  }

  translate(...args) {
    return lang.t(`error.${this.translation_key}`, ...args);
  }
}
```

## How do I test this PR?

- Login to a suspended user.
- Perform an unauthorized action like respecting.
- See that the mutation is rolled back successfully and that an error is logged to the console.
